src/components/DataCollector/createEvent.js

```diff
@@ -24,12 +26,21 @@ export default () => {
     mergeXdm: createMerger(content, "xdm"),
     mergeMeta: createMerger(content, "meta"),
     mergeQuery: createMerger(content, "query"),
+    documentUnloading() {
+      documentUnloading = true;
+    },
+    isDocumentUnloading() {
+      return documentUnloading;
+    },
     expectResponse() {
       expectsResponse = true;
```

Comment:
We could do some validation here and within documentUnloading() so that you can't expect a response when the document is unloading and vice versa.  I'm not sure if we would want to throw an error, log a warning, not allow you to, or have some kind of priority.  I haven't thought it through.